File: javascript/1460.Make-Two-Arrays-Equal-by-Reversing-Subarrays.js
/**
 * @param {number[]} target
 * @param {number[]} arr
 * @return {boolean}
 */
var canBeEqual = function (target, arr) {
  const map = new Map();

  for (const num of target) {
    map.set(num, (map.get(num) || 0) + 1);
  }

  for (const num of arr) {
    map.set(num, map.get(num) - 1);

    if (map.get(num) === 0) map.delete(num);
  }

  return map.size === 0;
};

File: python/1460.Make-Two-Arrays-Equal-by-Reversing-Subarrays.py
class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        freq = defaultdict(int)

        for n1, n2 in zip(target, arr):
            freq[n1] += 1
            freq[n2] -= 1

        for n in freq:
            if freq[n] != 0:
                return False

        return True